acados/ocp_nlp/ocp_nlp_ddp.c

```diff
@@ -0,0 +1,1514 @@
+                   qp_status, ddp_iter, qp_iter);
+#endif
+#if defined(ACADOS_WITH_OPENMP)
+            // restore number of threads
+            omp_set_num_threads(num_threads_bkp);
+#endif
+            if (nlp_opts->print_level > 1)
+            {
+                printf("\n Failed to solve the following QP:\n");
+                if (nlp_opts->print_level)
+                    print_ocp_qp_in(qp_in);
+            }
+
+            mem->status = ACADOS_QP_FAILURE;
+            mem->ddp_iter = ddp_iter;
+            mem->time_tot = acados_toc(&timer0);
+
+            return mem->status;
+        }
+
+        // Compute the optimal QP objective function value
+        nlp_mem->qp_cost_value = ocp_nlp_ddp_compute_qp_objective_value(dims, qp_in, qp_out,nlp_work, nlp_mem);
+
+        // Calculate step norm
+        // res_comp
+        mem->step_norm = 0.0;
+        double tmp_norm = 0.0;
+        for (int i = 0; i <= N; i++)
+        {
+            int sum = dims->nx[i]+dims->nu[i];
+            blasfeo_dvecnrm_inf(sum, &qp_out->ux[i], 0, &tmp_norm);
+            mem->step_norm = tmp_norm > mem->step_norm ? tmp_norm : mem->step_norm;
+        }
+        /* end solve QP */
+
+        /* globalization */
+        if (infeasible_initial_guess)
+        {
+            // Accept the forward simulation to get feasible initial guess
+            mem->alpha = 1.0;  // full step to obtain feasible initial gues
+            ocp_nlp_ddp_compute_trial_iterate(config, dims, nlp_in, nlp_out, nlp_opts, mem, nlp_work, mem->alpha);
+            copy_ocp_nlp_out(dims, work->nlp_work->tmp_nlp_out, nlp_out);
+            infeasible_initial_guess = false;
+        }
+        else
+        {
+            int linesearch_success = 0;
+            // Do the globalization here: Either fixed step or Armijo line search
+            acados_tic(&timer1);
+            // NOTE on timings: currently all within globalization is accounted for within time_glob.
+            //   QP solver times could be also attributed there alternatively. Cleanest would be to save them seperately.
+            if (nlp_opts->globalization == FIXED_STEP)
+            {
+                // Set the given step length
+                mem->alpha = nlp_opts->step_length;
+
+                // update variables
+                ocp_nlp_ddp_compute_trial_iterate(config, dims, nlp_in, nlp_out, nlp_opts, mem, nlp_work, mem->alpha);
+            }
+            else if (nlp_opts->globalization == MERIT_BACKTRACKING)
+            {
+                // do backtracking line search on objective function
+                linesearch_success = ocp_nlp_ddp_backtracking_line_search(config, dims, nlp_in, nlp_out, mem, work, opts);
+                evaluate_cost = false; // since the cost was already evaluated in the line search
+            }
+
+            mem->stat[mem->stat_n*(ddp_iter+1)+6] = mem->alpha;
+            // Copy new iterate to nlp_out
+            if (linesearch_success == 0)
```

Comment:
Here, I think the status, or more precisely the variable `linesearch_success` is a bit confusing. I would have expected a boolean meaning, i.e. 1 means success.
Of course, for main functions in C, 0 means success, but since `ocp_nlp_ddp_backtracking_line_search` is not a main function, I would prefer to change it to a boolean meaning.

Reply:
Sure :)